source/common/tracing/http_tracer_impl.h

```diff
@@ -187,5 +194,77 @@ class HttpTracerImpl : public HttpTracer {
   const LocalInfo::LocalInfo& local_info_;
 };
 
+class GeneralCustomTag : public CustomTag {
+public:
+  explicit GeneralCustomTag(const std::string& tag) : tag_(tag) {}
+  absl::string_view tag() const override { return tag_; }
+  void apply(Span& span, const CustomTagContext& ctx) const override;
+
+  virtual absl::string_view value(const CustomTagContext& ctx) const PURE;
+  virtual std::string toString() const PURE;
```

Comment:
AFAICT this is only used for testing. Please remove this and have the tests actually verify the created spans instead.